homeassistant/helpers/intent.py

```diff
@@ -252,11 +283,14 @@ def async_match_states(
     name: str | None = None,
     area_name: str | None = None,
     area: area_registry.AreaEntry | None = None,
+    floor_name: str | None = None,
+    floor: floor_registry.FloorEntry | None = None,
     domains: Collection[str] | None = None,
     device_classes: Collection[str] | None = None,
     states: Iterable[State] | None = None,
     entities: entity_registry.EntityRegistry | None = None,
     areas: area_registry.AreaRegistry | None = None,
+    floors: floor_registry.FloorRegistry | None = None,
```

Comment:
We used to have these, because it was an async task to get the registry. That's no longer the case so I think we can drop all of these from the params in a future PR